src/lib/eina/eina_thread.c

```diff
@@ -228,30 +122,42 @@ EAPI Eina_Bool
 eina_thread_cancel(Eina_Thread t)
 {
    if (!t) return EINA_FALSE;
-   return pthread_cancel((pthread_t)t) == 0;
+   return _eina_thread_cancel(t);
 }
 
 EAPI Eina_Bool
 eina_thread_cancellable_set(Eina_Bool cancellable, Eina_Bool *was_cancellable)
 {
-   int state = cancellable ? PTHREAD_CANCEL_ENABLE : PTHREAD_CANCEL_DISABLE;
+   int state = cancellable ? EINA_THREAD_CANCEL_ENABLE : EINA_THREAD_CANCEL_DISABLE;
    int old = 0;
    int r;
-
+   #ifndef _WIN32
    /* enforce deferred in case users changed to asynchronous themselves */
-   pthread_setcanceltype(PTHREAD_CANCEL_DEFERRED, &old);
+   
+   pthread_setcanceltype(EINA_THREAD_CANCEL_DEFERRED, &old);
```

Comment:
Who are those pthread functions working if you remover `#include <pthread.h>`?

Reply:
There is no more pthread_* in eina_thread.c